### PR TITLE
Add support for rdb

### PR DESCRIPTION
**Pull request by ashtul**

### CODE REVIEW

File: tests/pytests/test_vecsim.py [hunk 1]
```diff
@@ -296,12 +296,13 @@ def test_with_fields(env):
...
-    query_data = np.float32(np.random.random((1, dimension)))
-    res = env.cmd('FT.SEARCH', 'idx', '5 @v:[$vec_param TOPK 100]',
-                    'SORTBY', 'v_score', 'PARAMS', 2, 'vec_param', query_data.tobytes(),
-                    'RETURN', 2, 'v_score', 't')
-    res_nocontent = env.cmd('FT.SEARCH', 'idx', '5 @v:[$vec_param TOPK 100]',
-                    'SORTBY', 'v_score', 'PARAMS', 2, 'vec_param', query_data.tobytes(),
-                    'NOCONTENT')
-    env.assertEqual(res[1::2], res_nocontent[1:])
-    env.assertEqual('t', res[2][2])
+    for _ in env.retry_with_rdb_reload():
```
Comment (oshadmi) [inline tests/pytests/test_vecsim.py:299]: I suggest to add a DB with various VecSim data to short read test

File: src/spec.c [hunk 2]
```diff
@@ -1366,6 +1369,13 @@ static int FieldSpec_RdbLoad(RedisModuleIO *rdb, FieldSpec *f, int encver) {
     f->tagSep = *s;
     RedisModule_Free(s);
   }
+  // Load vector specific options
+  if (encver >= INDEX_VECSIM_VERSION ||
+      FIELD_IS(f, INDEXFLD_T_VECTOR) || (f->options & FieldSpec_Dynamic)) {
```
Comment (DvirDukhan) [inline src/spec.c:1374]: should it be the following?
```suggestion
  if (encver >= INDEX_VECSIM_VERSION &&
      (FIELD_IS(f, INDEXFLD_T_VECTOR) || (f->options & FieldSpec_Dynamic))) {
```

File: src/vector_index.c [hunk 3]
```diff
@@ -164,3 +165,47 @@ const char *VecSimAlgorithm_ToString(VecSimAlgo algo) {
...
+    RedisModule_SaveUnsigned(rdb, vecsimParams->bfParams.initialCapacity);
+    RedisModule_SaveUnsigned(rdb, vecsimParams->bfParams.blockSize);
+    break;
+  case VecSimAlgo_HNSWLIB:
+    RedisModule_SaveUnsigned(rdb, vecsimParams->hnswParams.initialCapacity);
+    RedisModule_SaveUnsigned(rdb, vecsimParams->hnswParams.M);
+    RedisModule_SaveUnsigned(rdb, vecsimParams->hnswParams.efConstruction);
+    RedisModule_SaveUnsigned(rdb, vecsimParams->hnswParams.efRuntime);
+    break;
+  }
```
Comment (DvirDukhan) [inline src/vector_index.c:186]: default fail?

File: src/vector_index.c [hunk 4]
```diff
@@ -164,3 +165,47 @@ const char *VecSimAlgorithm_ToString(VecSimAlgo algo) {
...
+}
+
+int VecSim_RdbLoad(RedisModuleIO *rdb, VecSimParams *vecsimParams, int encver) {
```
Comment (DvirDukhan) [inline src/vector_index.c:189]: why do you need the `encver` here?

File: src/vector_index.c [hunk 5]
```diff
@@ -164,3 +165,47 @@ const char *VecSimAlgorithm_ToString(VecSimAlgo algo) {
...
+    vecsimParams->bfParams.initialCapacity = LoadUnsigned_IOError(rdb, goto fail);
+    vecsimParams->bfParams.blockSize = LoadUnsigned_IOError(rdb, goto fail);
+    break;
+  case VecSimAlgo_HNSWLIB:
+    vecsimParams->hnswParams.initialCapacity = LoadUnsigned_IOError(rdb, goto fail);
+    vecsimParams->hnswParams.M = LoadUnsigned_IOError(rdb, goto fail);
+    vecsimParams->hnswParams.efConstruction = LoadUnsigned_IOError(rdb, goto fail);
+    vecsimParams->hnswParams.efRuntime = LoadUnsigned_IOError(rdb, goto fail);
+    break;
+  }
```
Comment (DvirDukhan) [inline src/vector_index.c:206]: default return RedisModule_ERR?

File: tests/pytests/test_vecsim.py (hunk 1 above)
Comment (DvirDukhan) [inline tests/pytests/test_vecsim.py:299]: why not have a dedicated function for rdb_reload? this is a reuse of an existing test case?

**Review by DvirDukhan — COMMENTED** [review]
Looks good, few comments

File: src/vector_index.c (hunk 5 above)
Comment (ashtul) [reply src/vector_index.c:206]: The enum has only 2 variants.
Using default disables warning at compilation

File: src/vector_index.c (hunk 3 above)
Comment (ashtul) [reply src/vector_index.c:186]: The enum has only 2 variants.
Using default disables warning at compilation

File: tests/pytests/test_vecsim.py (hunk 1 above)
Comment (ashtul) [reply tests/pytests/test_vecsim.py:299]: We should add more tests regardless
For this functionality, I think this is appropriate.

**Pull request merged**